google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineDeployAction.java

```diff
@@ -247,15 +241,72 @@ public void run(@NotNull ProgressIndicator indicator) {
 
   }
 
-  private String parseDeployOutputToModule(String jsonOutput)
+  // Just to return two strings.
+  static private class VersionService {
+    public String version = null;
+    public String service = null;
+  }
+
+  private VersionService parseDeployOutputToModule(String jsonOutput)
       throws JsonParseException {
-    Type deployOutputType = new TypeToken<Map<String, String>>() {}.getType();
-    Map<String, String> deployOutput = new Gson().fromJson(jsonOutput, deployOutputType);
+    VersionService versionService = new VersionService();
+
+    /* An example JSON output of gcloud app deloy:
+        {
+          "configs": [],
+          "versions": [
+            {
+              "id": "20160429t112518",
+              "last_deployed_time": null,
+              "project": "springboot-maven-project",
+              "service": "default",
+              "traffic_split": null,
+              "version": null
+            }
+          ]
+        }
+    */
 
-    if(deployOutput == null || deployOutput.keySet().size() != 1) {
-      throw new AssertionError("Expected a single module output from flex deployment.");
+    JsonReader reader = new JsonReader(new StringReader(jsonOutput));
+    try {
+      reader.beginObject(); // Top-level is a single object.
+
+      // Look for "versions" in the top-level object.
+      while (reader.hasNext()) {
+        String name = reader.nextName();
+
+        if ("versions".equals(name) && reader.peek() != JsonToken.NULL) {
+          reader.beginArray(); // "versions" should be an array.
+          // We hope the array has an element.
+          if (reader.hasNext()) {
+            reader.beginObject(); // An element in "versions" should be an object.
+
+            // Look for "id" and "service" inside the element.
+            while (reader.hasNext()) {
+              String nested_name = reader.nextName();
+              if ("id".equals(nested_name)) {
+                versionService.version = reader.nextString(); // "id" found. (String)
+              } else if ("service".equals(nested_name)) {
+                versionService.service = reader.nextString(); // "service" found. (String)
+              } else {
+                reader.skipValue(); // Skip a value if not "id".
+              }
+            }
+            reader.endObject();
```

Comment:
I feel like this will fail if "versions" contain multiple elements. I will need to fix this.